tests/publiccloud/run_ltp.pm

```diff
@@ -66,13 +66,15 @@ sub upload_ltp_logs
         my $parser  = OpenQA::Parser::Format::LTP->new()->load($log_file->to_string);
 
         if (find_whitelist_testsuite($ltp_testsuite)) {
+            my %ltp_log_results = map { $_->{test_fqn} => $_->{test} } @{$ltp_log->{results}};
             for my $result (@{$parser->results()}) {
-                next unless ($result->{result} ne 'ok');
-                my ($ltp_test) = grep { $_->{test_fqn} eq $result->{test_fqn} } @{$ltp_log->{results}};
-                for my $ret (@{$ltp_test->{test}->{retval}}) {
-                    if (override_known_failures($self, {%{$self->{ltp_env}}, result => $ret}, $ltp_testsuite, $result->{test_fqn})) {
+                # Mark only as softfail, if _all_ `retval` have a "known_issue" entry or the retval was 0
+                my @failed_attempts = grep { $_ ne 0 } @{$ltp_log_results{$result->{test_fqn}}->{retval}};
+                my @known_failures  = grep { find_whitelist_entry({%{$self->{ltp_env}}, result => $_}, $ltp_testsuite, $result->{test_fqn}) } @failed_attempts;
+
+                if (@failed_attempts > 0 && @failed_attempts == @known_failures) {
+                    if (override_known_failures($self, {%{$self->{ltp_env}}, result => shift @known_failures}, $ltp_testsuite, $result->{test_fqn})) {
```

Comment:
The code above is a bit too complicated. It'd be easier to find retvals that are *not* whitelisted like this:

    my @new_fails = grep { $_ ne 0 && !find_whitelist_entry(...) } @{$ltp_log_results{$result->{test_fqn}}->{retval}};

then call `override_known_failures()` if the array is empty. You don't need to check the return value then because you've already verified it'll be 1.

Also, the env variable is `retval`, not `result`.

Reply:
I would like to put the evaluation logic into `LTP::WhiteList` so it is done "always" the same way, WDYT about 1233dcbdaea626c464b4367528c77bf4a7ba3618

I have a problem with `$retval eq 0`. This actually makes the `override_known_failures()` behaves inconsistently see this https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13065/commits/49e87aff74ad3449fdd88bad0372ac28c2bfa659#diff-9f0479ba67450b0dd9dd303a261b4bcdef42036ff154db485b12dbde2600ad89R85 I would expect here, that we do not overwrite, like we do with `$retval = [ 0 ];`. 
But I would like to double check with you, if you also think we should return 0, if someone pass a environment with `$retval = 0;` to `override_known_failures()`. 
Or we keep it up to the caller and don't extract `0` from `$retval` if the retval is an array.

Also added a unit test for it :face_with_thermometer: